### PR TITLE
fix(react-chart): fix arrow displaying on corner point

### PR DESCRIPTION
**Pull request by Krijovnick**
Fixed only in MUI, in the bootstrap it is ok.
before:

after:

### CODE REVIEW

File: packages/dx-react-chart-material-ui/src/templates/tooltip/arrow.jsx [hunk 1]
```diff
@@ -50,8 +47,21 @@ const styles = (theme) => {
...
+
+BaseArrow.propTypes = {
+  placement: PropTypes.string.isRequired,
+  className: PropTypes.string,
+  classes: PropTypes.object,
+};
+
+BaseArrow.defaultProps = {
+  className: undefined,
+  classes: undefined,
```
Comment (MaximKudriavtsev) [inline packages/dx-react-chart-material-ui/src/templates/tooltip/arrow.jsx]: We often mark the `classes` property as a required

File: packages/dx-react-chart-material-ui/src/templates/tooltip/overlay.jsx [hunk 2]
```diff
@@ -18,25 +19,46 @@ const styles = (theme) => {
...
+  arrow: {
+    element: arrowRef,
+  },
+});
 
-export const Overlay = withStyles(styles)(({
+const OverlayBase = ({
   classes, className, children, target, rotated, arrowComponent: ArrowComponent, ...restProps
 }) => {
+  const [arrowRef, setArrowRef] = React.useState(null);
```
Comment (ushkal) [inline packages/dx-react-chart-material-ui/src/templates/tooltip/overlay.jsx]: `useRef` is better for this task

Comment (ushkal) [reply packages/dx-react-chart-material-ui/src/templates/tooltip/overlay.jsx]: I meant React.useRef hook :)

Comment (Krijovnick) [reply packages/dx-react-chart-material-ui/src/templates/tooltip/overlay.jsx]: If we use `useRef`, then we should use `useEffect` and it is not looks better.

File: packages/dx-react-chart-material-ui/src/templates/tooltip/overlay.jsx [hunk 3]
```diff
@@ -18,25 +19,46 @@ const styles = (theme) => {
...
+  classes: PropTypes.object,
+  children: PropTypes.node.isRequired,
+  target: PropTypes.any.isRequired,
+  rotated: PropTypes.bool.isRequired,
+  arrowComponent: PropTypes.oneOfType([PropTypes.func, PropTypes.object]).isRequired,
+};
+
+OverlayBase.defaultProps = {
+  className: undefined,
+  classes: undefined,
```
Comment (MaximKudriavtsev) [inline packages/dx-react-chart-material-ui/src/templates/tooltip/overlay.jsx]: Same with `classes` here

**Pull request merged**